app/assets/javascripts/services/subscription_service.js

```diff
@@ -0,0 +1,9 @@
+ManageIQ.angular.app.service('subscriptionService', ['$timeout', function($timeout) {
+  this.subscribeToEventType = function(eventType, callback) {
+    ManageIQ.angular.rxSubject.subscribe(function(event) {
+      if (event.eventType === eventType) {
+        $timeout(callback(event.response));
```

Comment:
This does not do what you want, unless `callback` is a function that returns a function.

Should be `$timeout(callback.bind({}, event.response);` or `$timeout(function() { callback(event.response); });`